crates/bevy_transform/src/hierarchy/hierarchy.rs

```diff
@@ -113,8 +135,20 @@ mod tests {
             .map(|(a, b)| (*a, *b))
             .collect::<Vec<_>>();
 
+        {
+            let children = world.get::<Children>(grandparent_entity).unwrap();
+            assert_eq!(
+                children.iter().any(|&i| i == parent_entity),
+                false,
+                "grandparent should no longer know about its child which has been removed"
+            );
+        }
+
         // parent_entity and its children should be deleted,
-        // the (0, 0) tuples remaining.
-        assert_eq!(results, vec![(0u32, 0u64), (0u32, 0u64), (0u32, 0u64)]);
+        // the grandparent tuple (1, 1) and (0, 0) tuples remaining.
+        assert_eq!(
+            results,
+            vec![(0u32, 0u64), (0u32, 0u64), (0u32, 0u64), (1u32, 1u64)]
```

Comment:
Question: Is the order of entities stable?  If not, then we should probably alter the grandparent to _also_ be `(0u32, 0u64)` to avoid a flaky test.